pkg/async/async.go

```diff
@@ -121,6 +121,8 @@ func Run(ctx context.Context, r Runner) {
 // RunBackground executes a single asynchronous task with background context
 //
 // It creates a new trace for the task and passes through deadlines.
+//
+// Deprecated: Please only use Run - you should always be aware of and control your context.
```

Comment:
```suggestion
// Deprecated: Use Run with context.Background() instead (if required).
```

Reply:
```suggestion
// Deprecated: Use Run with an explicit context.WithTimeout to prevent goroutine leak
```
Fine with the current wording, but if any thing I'd change to more like